### PR TITLE
Use enumeration to choose between hie, hls or ghcide

### LINKED ISSUE #222: Add a configuration option to switch easily between ghcide, hls and hie
* A select with fixed options could work
* We can take [emacs lsp-haskell as example](https://github.com/emacs-lsp/lsp-haskell/blob/e9c306b8246e2b50bf9260f71b3d2e5334b90b30/lsp-haskell.el#L39-L49)
* We should choose the initial default value (probably hie for now)
  * We could switch to hls at the right time
* Maybe the rest of params would need a redesign
* We can remove the actual shell scripts:
  * https://github.com/alanz/vscode-hie-server/blob/master/hie-vscode.bat
  * https://github.com/alanz/vscode-hie-server/blob/master/hie-vscode.sh

### PR DESCRIPTION
**Pull request by alanz**
closes #222 

I have tested on linux, switching between hls and hie.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -143,6 +116,20 @@
           "default": "",
           "description": "If set, redirects the logs to a file."
         },
+        "languageServerHaskell.hieVariant": {
+          "scope": "resource",
+          "type": "string",
+          "enum": ["haskell-ide-engine", "haskell-language-server", "ghcide"],
+          "default": "haskell-ide-engine",
+          "description": "Which hasell language server to use."
```
Comment (jneira) [inline package.json:124]:
```suggestion
          "description": "Which haskell language server to use."
```
typo

**Comment by jneira** [comment]
The server is warning about the "off" value:
```
[Error - 9:53:46] haskell-lsp:no handler for. Object (fromList [("jsonrpc",String "2.0"),("params",Object (fromList [("value",String "off")])),("method",String "$/setTraceNotification")])
```
But i think i've seen it before, it is not related with these changes.

File: src/extension.ts [hunk 2]
```diff
@@ -128,30 +118,33 @@ function activateHieNoCheck(context: ExtensionContext, folder: WorkspaceFolder,
...
-  const serverPath =
-    useCustomWrapper || hieExecutablePath ? hieLaunchScript : context.asAbsolutePath(path.join('.', hieLaunchScript));
+  const serverPath = hieLaunchScript;
 
-  const runArgs: string[] = [];
-  let debugArgs: string[] = [];
+  const runArgs: string[] = ['--lsp'];
+  let debugArgs: string[] = ['--lsp'];
   if (logLevel === 'messages') {
     debugArgs = ['-d'];
```
Comment (jneira) [inline src/extension.ts:145]:
```suggestion
    debugArgs = debugArgs.concat(['-d']);
```
hls crashes for me cause that line removed the `--lsp` arg

**Comment by alanz** [comment]
@jneira it looks like you made the necessary updates. What else needs to happen?

I was considering passing the intended exe into the call to `isHieInstalled`, which is why I added the parameter.

File: package.json [hunk 3]
```diff
@@ -143,6 +116,20 @@
           "default": "",
           "description": "If set, redirects the logs to a file."
         },
+        "languageServerHaskell.hieVariant": {
```
Comment (bubba) [inline package.json:119]: Nitpick, should probably name this server variant or something to be agnostic
```suggestion
        "languageServerHaskell.serverVariant": {
```

Comment (jneira) [reply package.json:119]: @alanz
> @jneira it looks like you made the necessary updates. What else needs to happen?

Not much more, i hope you didn't mind than i commited directly to the branch.
With the changes the extension worked for me in windows, switching between the three sever options and with both `-d`  and `-l` set.

> I was considering passing the intended exe into the call to `isHieInstalled`, which is why I added the parameter.

ok, i'll try to implement the change later

**Comment by jneira** [comment]
@alanz i'updated it to check the appropiate executable for each variant

**Pull request merged**